lib/hierarchy.js

```diff
@@ -112,8 +110,7 @@ module.exports = function (comments) {
           node.comments.length ? path.concat(node.comments[0]) : []);
       }
 
-      for (var i = 0; i < node.comments.length; i++) {
-        var comment = node.comments[i];
+      for (var comment of node.comments) {
```

Comment:
Your PR description didn't mention for-of? I'm fine with that too even though it is pretty slow in Node4